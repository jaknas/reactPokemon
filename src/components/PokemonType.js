import React from 'react';
import PropTypes from 'prop-types';
import { typeColors, style } from '../helpers/styles';

const PokemonType = ({ variant, groupBy }) => (
  <React.Fragment>
    {groupBy === 'li' && typeof variant !== 'undefined'
      ? variant.map(type => (
          <li
            key={type}
            className="list-group-item"
            style={{
              backgroundColor: typeColors[type],
              ...style,
            }}
          >
            {type}
          </li>
      ))
      : null}
    {groupBy === 'span'
      ? variant.map(type => (
          <span
            key={type}
            className="d-inline text-center badge"
            style={{
              backgroundColor: typeColors[type],
              ...style,
            }}
          >
            {type}
          </span>
      ))
      : null}
  </React.Fragment>
);
PokemonType.propTypes = {
  variant: PropTypes.array,
  groupBy: PropTypes.string.isRequired,
};

export default PokemonType;
