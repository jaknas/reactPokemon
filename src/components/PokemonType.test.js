import React from "react";
import { shallow } from "enzyme";
import PokemonType from "./PokemonType";

it("expect to render PokemonType component", () => {
	const mockPokemon = [
		{
			id: 1,
			type: [1, 2, 3]
		}
	];

	expect(shallow(<PokemonType pokemon={mockPokemon[0]} />)).toMatchSnapshot();
});
