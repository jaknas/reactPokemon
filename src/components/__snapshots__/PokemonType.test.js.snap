// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render PokemonType component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    pokemon={
        Object {
            "id": 1,
            "type": Array [
              1,
              2,
              3,
            ],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="d-inline text-center"
          style={
                    Object {
                              "backgroundColor": undefined,
                              "border": "1px solid rgba(0,0,0,0.2)",
                              "borderRadius": "10px",
                              "color": "white",
                              "margin": "2px",
                              "padding": "1px 8px 4px 8px",
                              "textShadow": "1px 1px 2px rgba(0,0,0,0.7)",
                            }
          }
>
          1
</span>,
        <span
          className="d-inline text-center"
          style={
                    Object {
                              "backgroundColor": undefined,
                              "border": "1px solid rgba(0,0,0,0.2)",
                              "borderRadius": "10px",
                              "color": "white",
                              "margin": "2px",
                              "padding": "1px 8px 4px 8px",
                              "textShadow": "1px 1px 2px rgba(0,0,0,0.7)",
                            }
          }
>
          2
</span>,
        <span
          className="d-inline text-center"
          style={
                    Object {
                              "backgroundColor": undefined,
                              "border": "1px solid rgba(0,0,0,0.2)",
                              "borderRadius": "10px",
                              "color": "white",
                              "margin": "2px",
                              "padding": "1px 8px 4px 8px",
                              "textShadow": "1px 1px 2px rgba(0,0,0,0.7)",
                            }
          }
>
          3
</span>,
      ],
      "className": "text-center",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": 1,
          "className": "d-inline text-center",
          "style": Object {
            "backgroundColor": undefined,
            "border": "1px solid rgba(0,0,0,0.2)",
            "borderRadius": "10px",
            "color": "white",
            "margin": "2px",
            "padding": "1px 8px 4px 8px",
            "textShadow": "1px 1px 2px rgba(0,0,0,0.7)",
          },
        },
        "ref": null,
        "rendered": 1,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": 2,
          "className": "d-inline text-center",
          "style": Object {
            "backgroundColor": undefined,
            "border": "1px solid rgba(0,0,0,0.2)",
            "borderRadius": "10px",
            "color": "white",
            "margin": "2px",
            "padding": "1px 8px 4px 8px",
            "textShadow": "1px 1px 2px rgba(0,0,0,0.7)",
          },
        },
        "ref": null,
        "rendered": 2,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": 3,
          "className": "d-inline text-center",
          "style": Object {
            "backgroundColor": undefined,
            "border": "1px solid rgba(0,0,0,0.2)",
            "borderRadius": "10px",
            "color": "white",
            "margin": "2px",
            "padding": "1px 8px 4px 8px",
            "textShadow": "1px 1px 2px rgba(0,0,0,0.7)",
          },
        },
        "ref": null,
        "rendered": 3,
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="d-inline text-center"
            style={
                        Object {
                                    "backgroundColor": undefined,
                                    "border": "1px solid rgba(0,0,0,0.2)",
                                    "borderRadius": "10px",
                                    "color": "white",
                                    "margin": "2px",
                                    "padding": "1px 8px 4px 8px",
                                    "textShadow": "1px 1px 2px rgba(0,0,0,0.7)",
                                  }
            }
>
            1
</span>,
          <span
            className="d-inline text-center"
            style={
                        Object {
                                    "backgroundColor": undefined,
                                    "border": "1px solid rgba(0,0,0,0.2)",
                                    "borderRadius": "10px",
                                    "color": "white",
                                    "margin": "2px",
                                    "padding": "1px 8px 4px 8px",
                                    "textShadow": "1px 1px 2px rgba(0,0,0,0.7)",
                                  }
            }
>
            2
</span>,
          <span
            className="d-inline text-center"
            style={
                        Object {
                                    "backgroundColor": undefined,
                                    "border": "1px solid rgba(0,0,0,0.2)",
                                    "borderRadius": "10px",
                                    "color": "white",
                                    "margin": "2px",
                                    "padding": "1px 8px 4px 8px",
                                    "textShadow": "1px 1px 2px rgba(0,0,0,0.7)",
                                  }
            }
>
            3
</span>,
        ],
        "className": "text-center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": 1,
            "className": "d-inline text-center",
            "style": Object {
              "backgroundColor": undefined,
              "border": "1px solid rgba(0,0,0,0.2)",
              "borderRadius": "10px",
              "color": "white",
              "margin": "2px",
              "padding": "1px 8px 4px 8px",
              "textShadow": "1px 1px 2px rgba(0,0,0,0.7)",
            },
          },
          "ref": null,
          "rendered": 1,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": 2,
            "className": "d-inline text-center",
            "style": Object {
              "backgroundColor": undefined,
              "border": "1px solid rgba(0,0,0,0.2)",
              "borderRadius": "10px",
              "color": "white",
              "margin": "2px",
              "padding": "1px 8px 4px 8px",
              "textShadow": "1px 1px 2px rgba(0,0,0,0.7)",
            },
          },
          "ref": null,
          "rendered": 2,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": 3,
            "className": "d-inline text-center",
            "style": Object {
              "backgroundColor": undefined,
              "border": "1px solid rgba(0,0,0,0.2)",
              "borderRadius": "10px",
              "color": "white",
              "margin": "2px",
              "padding": "1px 8px 4px 8px",
              "textShadow": "1px 1px 2px rgba(0,0,0,0.7)",
            },
          },
          "ref": null,
          "rendered": 3,
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
